L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelKalmanAlgo.cc

```diff
@@ -353,25 +358,41 @@ void L1TMuonBarrelKalmanAlgo::propagate(L1MuKBMTrack& track) {
     KNew = K + deltaK;
 
   //phi propagation
-  int phi11 = fp_product(aPhi_[step - 1], K, 10);
-  int phi12 = fp_product(-bPhi_[step - 1], phiB, 10);
+  ap_fixed<BITSCURV, BITSCURV> phi11 = ap_fixed<BITSPARAM + 1, 2>(aPhi_[step - 1]) * ap_fixed<BITSCURV, BITSCURV>(K);
+  ap_fixed<BITSPHIB, BITSPHIB> phi12 =
+      ap_fixed<BITSPARAM + 1, 2>(-bPhi_[step - 1]) * ap_fixed<BITSPHIB, BITSPHIB>(phiB);
+
   if (verbose_) {
-    printf("phi prop = %d* %f = %d, %d* %f =%d\n", K, aPhi_[step - 1], phi11, phiB, -bPhi_[step - 1], phi12);
+    std::cout << "phi prop = " << K << "*" << ap_fixed<BITSPARAM + 1, 2>(aPhi_[step - 1]) << " = " << phi11 << ", "
```

Comment:
You should avoid using `std::cout`, please use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger
(see https://github.com/cms-sw/cms-sw.github.io/issues/94).
Please fix it in all the file (18 times)

Reply:
Would reverting to printf() statements be ok?  We use this format to make hardware patterns that can be very expensive, so the activation of the logger would give a lot of output for no reason.